doc/source/whatsnew/v0.24.0.txt

```diff
@@ -1205,6 +1205,7 @@ Indexing
 ^^^^^^^^
 
 - The traceback from a ``KeyError`` when asking ``.loc`` for a single missing label is now shorter and more clear (:issue:`21557`)
+- :class:`PeriodIndex` now emits a ``KeyError`` when a malformed string is looked up, which is consistent with the behavior of :class:`DateTimeIndex` (:issue:`22803`)
```

Comment:
Are we calling this a bug fix or an API breaking change?

Reply:
Not sure there is a general rule, but given that it was undocumented, I'm lying in favor of the bug.

Reply:
i think this is ok here.